File: tests/java/TestAdapter.java
public class TestAdapter {
  public static void main(String[] args) {
    System.out.println("Testing Java Adapter...");
    try {
      Client.main(args);
    } catch (Exception e) {
      e.printStackTrace();
      System.exit(1);
    }
    System.out.println("Java Adapter Test Passed!");
  }
}

File: tests/c/TestAdapter.c
#define main pattern_main
#include "../../oop_in_c/Adapter/Adapter.c"
#undef main
#include <stdio.h>

int main() {
  printf("Testing C Adapter...\n");
  // Call pattern main
  pattern_main();
  printf("C Adapter Test Passed!\n");
  return 0;
}
